File: 3612. Process String with Special Operations I.js
/**
 * @param {string} inputString
 * @return {string}
 */
var processStr = function(inputString) {

    // Stores the processed string
    let processedString = "";

    for (const currentCharacter of inputString) {

        // If it is a lowercase letter,
        // append it to the result
        if (currentCharacter >= 'a' &&
            currentCharacter <= 'z') {

            processedString +=
                currentCharacter;
        }

        // Remove the last character
        else if (currentCharacter === '*') {

            if (processedString.length > 0) {

                processedString =
                    processedString.slice(
                        0,
                        processedString.length - 1
                    );
            }
        }

        // Duplicate the current string
        else if (currentCharacter === '#') {

            processedString +=
                processedString;
        }

        // Reverse the current string
        else if (currentCharacter === '%') {

            processedString =
                processedString
                    .split('')
                    .reverse()
                    .join('');
        }
    }

    return processedString;
};

File: 3612. Process String with Special Operations I.cpp
class Solution {
public:

    string processStr(string inputString) {

        // Stores the final processed string
        string processedString;

        for (char currentCharacter : inputString) {

            // If it is a lowercase letter,
            // append it to the result
            if (currentCharacter >= 'a' &&
                currentCharacter <= 'z') {

                processedString.push_back(
                    currentCharacter
                );
            }

            // Remove the last character
            else if (currentCharacter == '*') {

                if (!processedString.empty()) {

                    processedString.pop_back();
                }
            }

            // Duplicate the current string
            else if (currentCharacter == '#') {

                processedString +=
                    processedString;
            }

            // Reverse the current string
            else if (currentCharacter == '%') {

                reverse(
                    processedString.begin(),
                    processedString.end()
                );
            }
        }

        return processedString;
    }
};
